cluster/gce/gci/configure-helper.sh

```diff
@@ -83,16 +134,66 @@ function safe-format-and-mount() {
   mount -o discard,defaults "${device}" "${mountpoint}"
 }
 
-# Local ssds, if present, are mounted at /mnt/disks/ssdN.
+function unique-uuid-bind-mount(){
+  device=$1
+  interface=$2
+  mountpoint=$3
+
+  # Trigger udev refresh so that newly formatted devices are propagated in by-uuid
+  udevadm control --reload-rules
+  udevadm trigger
+  udevadm settle
+
+  # Get the real UUID of the device
+  item=$(readlink -f ${device} | cut -d '/' -f 3)
+  myuuid=$(ls -l /dev/disk/by-uuid/ | grep ${item} | tr -s ' ' | cut -d ' ' -f 9) 
+  bindpoint="${UUID_MNT_PREFIX}-${interface,,}-fs/local-ssd-${myuuid}"
+
+  # Mount device to the mountpoint
+  mkdir -p "${bindpoint}"
+  echo "Binding '${device}' at '${bindpoint}'"
+  mount --bind "${mountpoint}" "${bindpoint}"
+  chmod a+w ${bindpoint}
+
+}
+
+function mount-ext(){
+  ssd=$1
+  interface=$2
+  format=$3  
+
+  if [[ ${format,,} == "fs" ]]; then
+    #safe-format-and-mount-by-interface ${ssd} ${interface,,}
+    if [[ ${interface,,} == "scsi" ]]; then
+      scsinum=`echo ${ssd} | sed -e 's/\/dev\/disk\/by-id\/google-local-ssd-\([0-9]*\)/\1/'`
+      mountpoint="/mnt/disks/ssd${scsinum}"
+    else
+      nvmenum=`echo ${ssd} | sed -e 's/\/dev\/disk\/by-id\/google-nvme0n\([0-9]*\)/\1/'`
+      mountpoint="/mnt/disks/ssd-nvme${nvmenum}"
+    fi
+    safe-format-and-mount ${ssd} ${mountpoint}
+    unique-uuid-bind-mount ${ssd} ${interface} ${mountpoint}
+  elif [[ ${format,,} == "block" ]]; then
+    safe-block-symlink ${ssd} ${interface,,}
+  else
+    echo "Disk format must be either fs or block, got ${format}"
+  fi
+}
+
+# Local ssds, if present, are mounted TODO(dyzz) finish note
 function ensure-local-ssds() {
+  get-local-disk-num "scsi" "fs"
+  scsifsnum=${total}
+  scsifsnum=$((scsifsnum+NODE_LOCAL_SSDS))
```

Comment:
I think this is going to break GKE because GKE doesn't set NODE_LOCAL_SSDS, so scsifsnum will be 0.

Reply:
do they still provision localssds without setting that env var?

Reply:
Yes, GKE creates the instances differently than the cluster/gce/util.sh script.

Reply:
We may need to do something like:

1. First setup all the block devices
2. The remainder is for filesystem